src/runners/scilla_checker.ml

```diff
@@ -136,9 +142,11 @@ let check_lmodule file =
     let elibs = import_libs lmod.elibs in
     let%bind (recursion_lmod, recursion_rec_principles, recursion_elibs) = 
       check_recursion_lmod lmod elibs in
-    let%bind (typed_lmod, typed_elibs, typed_rlibs) = 
+    let%bind (typed_lmod, typed_rlibs, typed_elibs) = 
       check_typing_lmod recursion_lmod recursion_rec_principles recursion_elibs in
-    pure (typed_lmod, typed_elibs, typed_rlibs)
+    (* For checking name clashes of libraries, we also include _this_library. *)
+    let%bind _ = check_lib_nameclashes (typed_elibs @ [typed_lmod.libs]) in
```

Comment:
Don't we want to check for name clashes before typechecking? If there is a name clash with a function that is actually used, this could lead to incomprehensible type errors, and the name clashes would not be reported.